commands/audit/jas/secrets/secretsscanner.go

```diff
@@ -29,38 +32,45 @@ type SecretScanManager struct {
 // Return values:
 // []utils.IacOrSecretResult: a list of the secrets that were found.
 // error: An error object (if any).
-func RunSecretsScan(scanner *jas.JasScanner) (results []*sarif.Run, err error) {
-	secretScanManager := newSecretsScanManager(scanner)
-	log.Info("Running secrets scanning...")
-	if err = secretScanManager.scanner.Run(secretScanManager); err != nil {
+func RunSecretsScan(auditParallelRunner *utils.AuditParallelRunner, scanner *jas.JasScanner, extendedScanResults *utils.ExtendedScanResults,
+	module jfrogappsconfig.Module, threadId int) (err error) {
+	var scannerTempDir string
+	if scannerTempDir, err = jas.CreateScannerTempDirectory(scanner, string(utils.Secrets)); err != nil {
+		return
+	}
+	secretScanManager := newSecretsScanManager(scanner, scannerTempDir)
+	log.Info("[thread_id: " + strconv.Itoa(threadId) + "] Running secrets scanning...")
+	if err = secretScanManager.scanner.Run(secretScanManager, module); err != nil {
 		err = utils.ParseAnalyzerManagerError(utils.Secrets, err)
 		return
 	}
-	results = secretScanManager.secretsScannerResults
+	results := secretScanManager.secretsScannerResults
 	if len(results) > 0 {
-		log.Info("Found", utils.GetResultsLocationCount(results...), "secrets")
+		log.Info("[thread_id: "+strconv.Itoa(threadId)+"] Found", utils.GetResultsLocationCount(results...), "secrets")
 	}
+	auditParallelRunner.Mu.Lock()
+	extendedScanResults.SecretsScanResults = append(extendedScanResults.SecretsScanResults, results...)
+	auditParallelRunner.Mu.Unlock()
```

Comment:
this is not related to this logic, use it outside of this and return the result of the scan to be used outside.